Add submitting guidelines from grass-wiki (#2765)
OSGeo/grass: doc/development/submitting/docs.md

# Submitting Documentation

There are three types of documentation

> - **Library programmer's manual**: we [​use doxygen and document the functions](http://grass.osgeo.org/programming8/)

Reviewer: ```suggestion
- **Library programmer's manual**: we [​use doxygen and document the functions](https://grass.osgeo.org/programming8/)
```

  directly in the source code. See lib/gis/\*.c and lib/gis/gislib.dox for examples

- **User manual**: each command ("module") comes with its own page. We write it
  in simple HTML, storing the manual in a file `'<module>.html'` within the
  subdirectory of the module. See below for details on the structure.

- **Python documentation**: written in Markdown which is compiled with Sphinx

>   (see [​PyGRASS documentation](http://grass.osgeo.org/grass-stable/manuals/libpython/pygrass_index.html))

Reviewer: ```suggestion
  (see [​PyGRASS documentation](https://grass.osgeo.org/grass-stable/manuals/libpython/pygrass_index.html))
```

the figure name prefix should be the module name. See
[​raster/r.terraflow/r.terraflow.html](https://github.com/OSGeo/grass/blob/main/raster/r.terraflow/r.terraflow.html)
([manual](https://grass.osgeo.org/grass-stable/manuals/r.terraflow.html)) for
an example.

A number of major sections should be present in each help page.

R = Required
S = Suggested
O = Optional

In recommended order (in bold the required entries):

```html
**R: <h2>DESCRIPTION</h2>**
S: <h2>NOTE</H2>, <h2>NOTES</h2>
S: <h2>EXAMPLE</h2>, <h2>EXAMPLES</h2>
O: <h2>TODO</h2>
O: <h2>KNOWN ISSUES</h2>
O: <h2>REFERENCE</h2>, <h2>REFERENCES</h2>
**R: <h2>SEE ALSO</h2>**
**R: <h2>AUTHOR</h2>, <h2>AUTHORS</h2>**
```

Note that the parameter manual section is auto-generated upon compilation. This
is done by running the module in a virtual session after compilation (see the
output of 'make'). To subsequently verify the final HTML page, check the
resulting HTML pages which will be stored with the name of the module.

Examples (please add some more) should be coded like this:

```html
<div class="code"><pre>
v.to.db map=soils type=area option=area column=area_size unit=h
</pre></div>
```

> The [​online WWW manual pages](http://grass.osgeo.org/grass-stable/manuals/) are

Reviewer: ```suggestion
The [​online WWW manual pages](https://grass.osgeo.org/grass-stable/manuals/) are
```

<div class="code"><pre>
v.to.db map=soils type=area option=area column=area_size unit=h
</pre></div>
```

The [​online WWW manual pages](http://grass.osgeo.org/grass-stable/manuals/) are
updated every Saturday (from GitHub repository).

### Supported HTML Tags

Since the MAN conversion of g.html2man is limited, please use no other HTML
tags than:

```html
<a> <b> <body> <br> <code> <dd> <dl> <dt> <em>
<h2> <h3> <h4> <head> <hr> <i> <img> <li> <ol> <p>
<pre> <sup> <table> <td> <th> <title> <tr> <ul>
```

- Note that all tags have a closing tag except for `<hr>`, `<br>` and `<p>`.
- Use lower case forms.
- Do not insert `<p>` after `<h2>...</h2>` or `<h3>...</h3>`

Note that HTML is converted to MAN pages by [tools/g.html2man/](https://trac.osgeo.org/grass/browser/grass/trunk/tools/g.html2man)

### Markup style guide

- Module names (i.e., v.category) should be emphasized with `<em>module</em>`.
- Flags and parameter names written in boldface like `<b>-f</b>` and `<b>input</b>`.
- Shell commands, names, values, etc. should use `<tt>42</tt>`.
- Emphasized phrases `<i>should use italics</i>`.

The `<h2>SEE ALSO</h2>` section of each page should also be alphabetized.

### Footer

> Change in 2019: no longer useful after migration to git:
> 
> ~~The last line of the page should be the date of the last modification of the
> page which is updated automatically by Subversion (during `svn commit`). In
> other words, include the following example line as is. It will be updated to
> the correct date automatically.~~
> 
> ```html
> <p><i>Last changed: $Date: 2014-08-14 11:34:43 -0400 (Thu, 14 Aug 2014) $</i>
> ```

Reviewer: ```suggestion

No footer to be added, since it will be added automatically during compilation.
```

### Break long lines

> To avoid git merge conflicts, please break a line at approximately 70-80 chars.

Reviewer: ```suggestion
In order to minimize potential git merge conflicts, please break a line at approximately 70-80 chars.
```
